docs/catalog-data/advanced_data_catalog_usage.md

```diff
@@ -225,4 +298,85 @@ catalog.save("test_dataset", data1)  # emits a UserWarning due to version incons
 # raises DatasetError since the data/01_raw/test.csv/exact_load_version/test.csv
 # file does not exist
 reloaded = catalog.load("test_dataset")
+
+### How to access dataset patterns
+
+The pattern resolution logic in `DataCatalog` is handled by the `config_resolver`, which can be accessed as a property of the catalog:
+
+```python
+config_resolver = catalog.config_resolver
+ds_config = catalog.config_resolver.resolve_pattern(ds_name)  # Resolve specific pattern
+patterns = catalog.config_resolver.list_patterns() # List all patterns
+```
+
+!!! note
+    `DataCatalog` does not support all dictionary methods, such as `pop()`, `popitem()`, or `del`.
+
+## How to save catalog to config
+
+You can serialize a `DataCatalog` into configuration format (e.g., for saving to a YAML file) using `.to_config()`:
+
+```python
+from kedro.io import DataCatalog
+from kedro_datasets.pandas import CSVDataset
+
+cars = CSVDataset(
+     filepath="cars.csv",
+     save_args={"index": False}
+ )
+catalog = DataCatalog(datasets={'cars': cars})
+
+config, credentials, load_versions, save_version = catalog.to_config()
+```
+
+To reconstruct the catalog later:
+```python
+new_catalog = DataCatalog.from_config(config, credentials, load_versions, save_version)
+```
+
+!!! note
+    This method only works for datasets with static, serializable parameters. For example, you can serialize credentials passed as dictionaries, but not as actual credential objects (like `google.auth.credentials.Credentials)`. In-memory datasets are excluded.
+
+## How to filter catalog datasets
+
+Use the `.filter()` method to retrieve dataset names that match specific criteria:
+
+```python
+import re
+from kedro.io import MemoryDataset
+from kedro_datasets.pandas import SQLQueryDataset
+
+catalog.filter(name_regex="raw")  # Names containing 'raw'
+catalog.filter(name_regex=re.compile("^model_"))  # Regex match (precompiled)
+catalog.filter(type_regex="pandas.excel_dataset.ExcelDataset")  # Match by type string
+catalog.filter(name_regex="train", type_regex="CSV")  # Name + type
+catalog.filter(name_regex="data", by_type=SQLQueryDataset)  # Exact type match
+catalog.filter(name_regex="data", by_type=[MemoryDataset, SQLQueryDataset])  # Multiple types
+```
+
+**Args:**
+
+- name_regex: Dataset names to match (string or re.Pattern).
+- type_regex: Match full class path of dataset types.
+- by_type: Dataset class(es) to match using isinstance.
+
+**Returns:**
+
+- A list of matching dataset names.
```

Comment:
Isn't this a bit redundant with what's automatically documented in https://docs.kedro.org/en/develop/api/io/kedro.io.DataCatalog/#kedro.io.DataCatalog.filter ?